### PR TITLE
Added travis file + fixed sockjs dependency on npm 2.x

### PR DESCRIPTION
**Pull request by siddharthkp**
There is no need for adding `scripts` in the travis file, it will run `npm install` and `npm test` by default.

```yml
language: node_js
```

Node versions: `[current, 6, 4]` based on [Node LTS plan](https://github.com/nodejs/LTS)

```yml
node_js:
  - "node"
  - "6"
  - "4"
```

Caching `node_modules` for faster builds

```yml
cache:
  directories:
    - node_modules
```

Link to results - https://travis-ci.org/siddharthkp/next.js/jobs/173480820

---

On [client/webpack-dev-client/socket.js#L1](https://github.com/zeit/next.js/blob/master/client/webpack-dev-client/socket.js#L1), there is

```js
import SockJS from 'sockjs-client'
```
`sockjs-client` is not declared as a dependency in `package.json`. `webpack-dev-server` however, has `sockjs-client` as a dependency.

On npm 3.x, this piece of code works because of the flat structure of `node_modules`

But on npm 2.x, it breaks because the correct path is `webpack-dev-server/node_modules/sockjs-client` To fix this, added `sockjs-client` to `dependencies`.

<details>
  <summary>Build step was failing on npm 2.x</summary>

link to travis - https://travis-ci.org/siddharthkp/next.js/builds/173474914

```
stream.js:74
      throw er; // Unhandled stream error in pipe.
      ^
Error: ./dist/client/webpack-dev-client/socket.js
Module not found: Error: Cannot resolve module 'sockjs-client' in /Users/travis/build/siddharthkp/next.js/dist/client/webpack-dev-client
resolve module sockjs-client in /Users/travis/build/siddharthkp/next.js/dist/client/webpack-dev-client
  looking for modules in /Users/travis/build/siddharthkp/next.js/node_modules
    /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client doesn't exist (module as directory)
    resolve 'file' sockjs-client in /Users/travis/build/siddharthkp/next.js/node_modules
      resolve file
        /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client doesn't exist
        /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.webpack.js doesn't exist
        /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.web.js doesn't exist
        /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.js doesn't exist
        /Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.json doesn't exist
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client]
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client]
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.webpack.js]
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.web.js]
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.js]
[/Users/travis/build/siddharthkp/next.js/node_modules/sockjs-client.json]
 @ ./dist/client/webpack-dev-client/socket.js 8:20-44
```
</details>

### CODE REVIEW

**Comment by siddharthkp** [comment]
Closes https://github.com/zeit/next.js/issues/94

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: node_js
+node_js:
+  - "node"
+  - "6"
+  - "4"
+cache:
+  directories:
+    - node_modules
```
Comment (siddharthkp) [inline .travis.yml]: Caching `node_modules` for speed

**Comment by siddharthkp** [comment]
@nkzawa Not sure if `sockjs-client` should go in `devDependencies` or `dependencies`. Help?

**Comment by sedubois** [comment]
What are the `sockjs-client` and `ava` dependencies for?

**Comment by siddharthkp** [comment]
@sedubois Check out the error in https://travis-ci.org/siddharthkp/next.js/jobs/173474925

`Cannot resolve module 'sockjs-client' in /next.js/dist/client/webpack-dev-client`

**Comment by siddharthkp** [comment]
@sedubois I think adding `sockjs-client` is probably not the correct solution. It does exist inside `webpack-dev-client/node_modules`. Probably the path needs to be fixed.

**Comment by siddharthkp** [comment]
@sedubois 

This is the line I'm talking about - [client/webpack-dev-client/socket.js#L1](https://github.com/zeit/next.js/blob/master/client/webpack-dev-client/socket.js#L1)

```js
import SockJS from 'sockjs-client'
```
`sockjs-client` is not declared as a dependency in `package.json`. `webpack-dev-server` however, has `sockjs-client` as a dependency.

On npm 3.x, it will work because of the flat structure of `node_modules`
But on npm 2.x, it will break, because the correct path is `webpack-dev-server/node_modules/sockjs-client`

**Comment by nkzawa** [comment]
@siddharthkp good catch about `sockjs-client` !

**Pull request merged**